extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +91,57 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+
+    // Because sddefault.jpg isn't available for all videos, we try to load
+    // it and fallback to hqdefault.jpg.
+    loadPromise(imgPlaceholder).then(function waitTillReady() {
+      // Wait until the image has completed.
+      if (!imgPlaceholder.complete) {
+        return timer.promise(10).then(waitTillReady);
```

Comment:
Ok. That does give some ideas. Let's try this instead of polling:
```
loadPromise(imagePlaceholder).then(() => {
  // Poll until complete is set.
  if (!imgPlaceholder.complete) {
    return whenDocumentReady(this.getWin().document);
  }
}).then(() => {
  // process
})...
```

Assumption is that `complete` is not set until DOMReady. If that's the case we should definitely NOT do polling, so this is a good clue.